lib/src/linking.rs

```diff
@@ -13,11 +13,42 @@ use {
     wasmtime_wasi_nn::WasiNnCtx,
 };
 
+#[derive(Default)]
+pub struct Limiter {
+    /// Total memory allocated so far.
+    pub memory_allocated: usize,
+}
+
+impl wasmtime::ResourceLimiter for Limiter {
+    fn memory_growing(
+        &mut self,
+        current: usize,
```

Comment:
Can we confirm that this starts at zero? In other words, is `memory_growing` called when the initial heap for an instance is set up?

Reply:
The interface for allocating new `Memory` structures in wasmtime is [`Memory::new_dynamic`](https://github.com/bytecodealliance/wasmtime/blob/4f1b38f0223843bc48a1f6b1a0af0e9f154d00a0/crates/runtime/src/memory.rs#L672) and [`Memory::new_static`](https://github.com/bytecodealliance/wasmtime/blob/4f1b38f0223843bc48a1f6b1a0af0e9f154d00a0/crates/runtime/src/memory.rs#L689). Both dispatch to [`Memory::limit_new`](https://github.com/bytecodealliance/wasmtime/blob/4f1b38f0223843bc48a1f6b1a0af0e9f154d00a0/crates/runtime/src/memory.rs#L723), which will call `memory_growing` with a [`current` value of `0`](https://github.com/bytecodealliance/wasmtime/blob/4f1b38f0223843bc48a1f6b1a0af0e9f154d00a0/crates/runtime/src/memory.rs#L792).

Reply:
Awesome, thank you for checking!